### PR TITLE
Update test

### PR DESCRIPTION
**Pull request by tmmsartor**
Streamline and update test framework.
TODO: Add all supported routines:
- blasfeo_api lq and qr
- blasfeo_api blas2 blas1
- blas_api trmm trsm syrk

### CODE REVIEW

File: auxiliary/x_aux_lib.c [hunk 1]
```diff
@@ -380,7 +380,7 @@ void GEAD_LIBSTR(int m, int n, REAL alpha, struct MAT *sA, int ai, int aj, struc
 
 
 // set all elements of a strmat to a value
-void GESE(int m, int n, REAL alpha, struct MAT *sA, int ai, int aj)
+void GESE_LIBSTR(int m, int n, REAL alpha, struct MAT *sA, int ai, int aj)
```
Comment (giaf) [inline auxiliary/x_aux_lib.c]: why?

**Comment by giaf** [comment]
there is a conflict in Makefile.blas (I added REF_BLAS_SYSTEM to use -lblas --lapack)

File: auxiliary/x_aux_lib.c (hunk 1 above)
Comment (tmmsartor) [reply auxiliary/x_aux_lib.c]: To be consistent with the naming. But actually the naming convention for macro seems still in the process to be updated.
To recap:
- `GESE` style is used for functions with blas interface. I.e:
`#define COPY dcopy_`  in blasfeo_api/d_blas2.c
`void GEMM(`  in blas_api/xgemm_ref.c

- `GESE_LIBSTR` style for  blasfeo interface. I.e:
`#define TRMV_LNN_LIBSTR blasfeo_dtrmv_lnn` in blasfeo_api/d_blas2_lib.c 
But is not used in blasfeo_api/x_blas3_lib.c where `*_LIBSTR`, `STRMAT` are dropped in favor of `GEMM_NT` (routine name plus blas hard-coded flags), `XMAT`.

The plan is to make the blas3 current style the default naming convention for macro? 
In this case I propose to use `BLASFEO_GEMM_NT` for function with "blasfeo" interface, or in general having some way to distinguish a BLASFEO macro even with there are no hard-coded flags i.e. `GESE_`

**Comment by tmmsartor** [comment]
I Fixed the conflicts with the latest master.
The question about the macro naming convention is still open,
currently we use a mix of two conventions, namely STRMAT/LIBSTR and XMAT/_.
I vote for using:
- `BLAS_API_<(capitalized)routine name>`: for functions that operate on array (at the signature level)  
- `BLASFEO_API_<(capitalized)routine name>_<(capitalized)flags code>`, for functions that operate on blasfeo structures and XMAT, XVEC for those structures.

File: auxiliary/x_aux_lib.c (hunk 1 above)
Comment (giaf) [reply auxiliary/x_aux_lib.c]: Yes the plan was to adopt the blas3 naming convention as the default one. None of the macro name is exposed, so it is not such a high priority. Is it even necessary to distinguish between BLAS and BLASFEO API in the macro names?

**Comment by giaf** [comment]
Your naming proposal looks long :p

**Pull request merged**